### ISSUE TITLE
Does it support other labels than {0, 1}?

### ISSUE THREAD

**Issue by jiangbian22** [tqchen/xgboost#164]
I am applying xgboost on a ranking problem. And, I am wondering if xgboost support other labels, say any positive integer, than {0, 1}? Sorry, I did not find the corresponding instructions on the document.

**Comment by tqchen** [comment]
Yes, if you use pairwise rank, you can use arbitrary integer as rank levels. Note that though in the evaluation meaure, only ndcg is rank level awared

**Comment by jiangbian22** [comment]
Thanks, Tianqi!
One more question, can I use arbitrary integer in regression problem?

**Comment by tqchen** [comment]
Yes you can

**Comment by tqchen** [comment]
However, it is recommended to scale the output to some bounded values, for example [0,1] will allows you to use logistic regression besides the linear one

**Comment by jiangbian22** [comment]
Got it~ That's quite helpful!

**Comment by jiangbian22** [comment]
Hi Tianqi,

For logistic regression, if I use real values under the scale of [0, 1] as labels in both training and test data, is the auc metric computed by xgboost the weighted one based on the real value labels?

Thanks!

**Comment by tqchen** [comment]
It is based on the real value labels. The real value serves as a rank ground truth for AUC evaluation
